shutdown.go

```diff
@@ -0,0 +1,65 @@
+package gokit
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+)
+
+// Server will create a http.Server from the Go standard library
+type Server struct {
+	Addr    string        // TCP address to listen on, ":http" if empty
+	Handler http.Handler  // handler to invoke, http.DefaultServeMux if nil
+	Timeout time.Duration // Timeout to wait for shutdown
+}
+
+// ListenAndServe mirrors the function from the Go standard library
+func (s *Server) ListenAndServe() string {
+	server := http.Server{
+		Addr:    s.Addr,
+		Handler: s.Handler,
+	}
+
+	// We want to report the listener is closed.
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	var svrStr string
+	// Start the listener.
+	go func() {
+		svrStr = fmt.Sprintf("server : %s", server.ListenAndServe())
```

Comment:
Is it useful to have this string? We'll only see this when the server ends right?

Reply:
Yes, and this string records _how_ the server ended. Could be useful for logging purposes.